lib/utils/formatting.ts

```diff
@@ -0,0 +1,8 @@
+export function dasherize(str: string) {
```

Comment:
```suggestion
export function keepKebabOrSnakeCase(str: string) {
```

maybe this name describes better?

---

```js
// kebab
dasherize('a-b-c') === 'a-b-c',
// snake
dasherize('a_b_c') === 'a_b_c',

dasherize('a b c') === 'a-b-c',
dasherize('a-b c') === 'a-b-c',
dasherize('a_b c') === 'a_b-c',
dasherize('a-b_c') === 'a-b_c',
```

Reply:
I like the name, but the function transforms, right , I feel this needs to be showed in the name.

if you have camelCase it is transformed... hm... maybe something generic like transformCamelCase ? or transformName ? 

Honestly I don't really like my suggestions, but we deffinetly need to adjust the function name.

Reply:
I guess you could choose any other than just `dasherize`, and write JSDoc comment to describe its for now

```js
transformToKebabOrSnakeCase('fooBar') === 'foo-bar' // camel
transformToKebabOrSnakeCase('FooBar') === 'foo-bar' // pascal
transformToKebabOrSnakeCase('foo bar') === 'foo-bar' // spaced

// or maybe 'normalizeToKebabOrSnakeCase'
```

Reply:
I really liked including 'normalize' in the name, nice suggestion.

I've updated the code. I am not really confident in the jsdoc description, but we can always adjust it.